GetMessages/handler.ts

```diff
@@ -46,18 +61,18 @@ export function GetMessagesHandler(
   messageModel: MessageModel
 ): IGetMessagesHandler {
   return async fiscalCode => {
-    const retrievedMessagesIterator = messageModel.findMessages(fiscalCode);
-    const validMessagesIterator = filterResultIterator(
-      retrievedMessagesIterator,
-      // isPending is true when the message has been received from the sender
-      // but it's still being processed
-      message => message.isPending !== true
-    );
-    const publicExtendedMessagesIterator = mapResultIterator(
-      validMessagesIterator,
-      retrievedMessageToPublic
-    );
-    return ResponseJsonIterator(publicExtendedMessagesIterator);
+    return messageModel
+      .findMessages(fiscalCode)
+      .map(flattenAsyncIterator)
+      .map(_ => filterAsyncIterator(_, isRight))
+      .map(_ => mapAsyncIterator(_, e => e.value))
+      .map(_ => filterAsyncIterator(_, RetrievedNotPendingMessage.is))
```

Comment:
not?

Reply:
`RetrievedNotPendingMessage.is` is a guard, otherwise I had to implement `e => !e.isPending` which kinda lose type safety